datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -4123,4 +4127,34 @@ mod tests {
         "
         )
     }
+
+    /// Create a test table scan with uppercase column names for case sensitivity testing
+    fn test_table_scan_with_uppercase_columns() -> Result<LogicalPlan> {
+        let schema = Schema::new(vec![
+            Field::new("A", DataType::UInt32, false),
+            Field::new("B", DataType::UInt32, false),
+            Field::new("C", DataType::UInt32, false),
+        ]);
+        table_scan(Some("test"), &schema, None)?.build()
+    }
+
+    #[test]
+    fn filter_agg_case_insensitive() -> Result<()> {
+        let table_scan = test_table_scan_with_uppercase_columns()?;
```

Comment:
If the table also has a column, what'll happen?

Reply:
Great question, just tried this and it works as expected for both uppercase and lower case col, even if both are present in the schema at the same time. I added another test,  lmk if we should keep it or it's overkill.

Reply:
Also happy to see a related sqllogictest